controller/api/v1alpha1/notificationrouter_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NotificationRouterSpec defines the desired state of NotificationRouter
+type NotificationRouterSpec struct {
+	// Name the name of this router, if not provided NotificationRouter's own name from NotificationRouter.ObjectMeta will be used.
+	// +kubebuilder:validation:Optional
+	Name *string `json:"name,omitempty"`
+
+	// Events the events to trigger, or use * for any
+	Events []string `json:"events,omitempty"`
+
+	// Filters filters by object type
+	// +kubebuilder:validation:Optional
+	Filters []RouterFilterAttributes `json:"filters,omitempty"`
```

Comment:
Is there a reason we're calling this `RouterFilterAttributes` instead of just `RouterFilters`?  I feel like the latter is more in-line w/ other types (although doesn't matter from a yaml perspective)

Reply:
it doesn't matter, I copied it from the API in a hurry, I have changed it